RevenueCatUI/Templates/Components/Packages/PackageGroup/PackageGroupComponentViewModel.swift

```diff
@@ -34,12 +36,57 @@ public class PackageGroupComponentViewModel {
         self.component = component
         self.offering = offering
 
+        let info = try Self.getPackages(component: component, offering: offering)
+        self.defaultPackage = info.defaultPackage
+
         self.stackComponentViewModel = try self.component.toStackComponentViewModel(
+            components: info.availablePackageComponents.map { .package($0) },
             localizedStrings: localizedStrings,
             offering: offering
         )
     }
 
+    static func getPackages(
+        component: PaywallComponent.PackageGroupComponent,
+        offering: Offering
+    ) throws -> (defaultPackage: Package, availablePackageComponents: [PaywallComponent.PackageComponent]) {
+
+        // Get list of available package components and their packages
+        let availablePackageInfos = component.packages.compactMap { packageComponent in
+            let pkg = offering.availablePackages.first(where: { $0.packageIdentifier == packageComponent.packageID })
+            if let pkg {
+                return (component: packageComponent, package: pkg)
+            } else {
+                return nil
+            }
+        }
+
+        // We need packages
+        guard let firstPackage = availablePackageInfos.first else {
+            Logger.error(Strings.paywall_could_not_find_any_packages)
+            throw PackageGroupValidationError.noAvailablePackages("No available packages found")
```

Comment:
Is this an error scenario where the packages configured for this component are not available in the offering? If so, maybe the error message can reflect that? Something like: 

> Please make sure the packages configured for this package group are available in the Offering. 
  
(Can probably be improved with more context haha.)

Reply:
It may also happen if the products are not found in Apple and so on... Basically the "No offerings found" error all over again 😅. But yeah, maybe we can have a link with more information about the reasons for this error.

Reply:
Oh right, sad. 😢 Yea maybe for now we can link to Sharif's community post? And when we do the Offerings-are-empty project we make sure we cover this path?

Reply:
Yup I think we should do that 👍

Reply:
> It may also happen if the products are not found in Apple and so on

Yeah yeah, this is the main use case this is for ☝️ 😬